### PR TITLE
GLB - Module tests and fixes

### PR DESCRIPTION
**Pull request by LucaPrete**

### CODE REVIEW

File: modules/net-xlb/ssl_certificates.tf [hunk 1]
```diff
@@ -15,26 +15,38 @@
...
+  # HC is defined, create also a default HC
+  ssl_certificates_config = (
+    try(var.target_proxy_https_config.ssl_certificates, null) == null
+    || length(coalesce(try(var.target_proxy_https_config.ssl_certificates, null), [])) == 0
+    ? merge(
+      { default = var.ssl_certificates_config_defaults },
+      coalesce(var.ssl_certificates_config, {})
+    )
+    : coalesce(var.ssl_certificates_config, {})
+  )
```
Comment (juliocc) [inline modules/net-xlb/ssl_certificates.tf:28]: This feels unnecessarily complicated. For example `length(coalesce(try(var.target_proxy_https_config.ssl_certificates, null), [])) == length(try(var.target_proxy_https_config.ssl_certificates, []))`.

And I think you can probably simplify it a bit considering both parts of the ternary operator include `coalesce(var.ssl_certificates_config, {})`. Perhaps something like 
```
ssl_certificates_config = merge(coalesce(var.ssl_certificates_config, {}), {
  default = var.ssl_certificates_config_defaults
})
```

File: modules/net-xlb/target_proxy.tf [hunk 2]
```diff
@@ -19,16 +19,21 @@ locals {
...
-      for cert in try(var.target_proxy_https_config.ssl_certificates, ["default"]) :
+    try(var.target_proxy_https_config.ssl_certificates, null) == null
+    || length(coalesce(try(var.target_proxy_https_config.ssl_certificates, null), [])) == 0
+    ? try(
+      [google_compute_managed_ssl_certificate.managed["default"].id],
+      [google_compute_ssl_certificate.unmanaged["default"].id],
+      null
+    )
+    : [
+      for cert in try(var.target_proxy_https_config.ssl_certificates, []) :
```
Comment (juliocc) [inline modules/net-xlb/target_proxy.tf:30]: same as before.

**Review by juliocc — COMMENTED** [review]
nice tests. left a comment around some locals

**Comment by LucaPrete** [comment]
Hi @juliocc thanks again for the review and the clarifications today.
The two required blocks have been simplified.
Anyway, I decided to keep -instead- the HC part as it was, given with the simplification we were hitting different errors while executing tests for corner cases.

**Comment by juliocc** [comment]
LGTM. Thanks for the changes.

**Review by juliocc — APPROVED** [review]
